navigation/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/NavigationEngine.java

```diff
@@ -5,27 +5,27 @@
 import android.support.annotation.NonNull;
 import android.text.TextUtils;
 
-import com.mapbox.services.android.navigation.v5.listeners.AlertLevelChangeListener;
 import com.mapbox.services.android.navigation.v5.listeners.OffRouteListener;
 import com.mapbox.services.android.navigation.v5.listeners.ProgressChangeListener;
+import com.mapbox.services.android.navigation.v5.milestone.Milestone;
+import com.mapbox.services.android.navigation.v5.milestone.MilestoneEventListener;
+import com.mapbox.services.android.telemetry.utils.MathUtils;
 import com.mapbox.services.api.directions.v5.models.DirectionsRoute;
 import com.mapbox.services.commons.models.Position;
 
 import java.util.concurrent.CopyOnWriteArrayList;
 import java.util.concurrent.TimeUnit;
 
 /**
- * The navigation engine makes use of {@link UserOffRouteState}, {@link AlertLevelState}, etc. to first create a new
- * route progress object and then to invoke the appropriate callbacks depending on the navigation state.
- *
  * @since 0.2.0
  */
 class NavigationEngine {
 
   // Listeners
-  private CopyOnWriteArrayList<AlertLevelChangeListener> alertLevelChangeListeners;
   private CopyOnWriteArrayList<ProgressChangeListener> progressChangeListeners;
   private CopyOnWriteArrayList<OffRouteListener> offRouteListeners;
+  private CopyOnWriteArrayList<MilestoneEventListener> milestoneEventListeners;
+  private CopyOnWriteArrayList<Milestone> milestones;
```

Comment:
@cammace Sorry, I wasn't clear. It's fine that we have two different lists. What I'm wondering is if there's a way to avoid having to duplicate both lists in two different classes (`MapboxNavigation` and `NavigationEngine`).